### PR TITLE
[RFC] SOTA vertical subsystem description

### PR DESCRIPTION
**Pull request by FlorentRevest**
This is my first attempt at detailing the basic concepts behind SOTA for an automotive platform. This might need a couple of additional iterations depending on the comments you have. The details of my actual researches are to be found in another RFC PR in the software-factory repository.

### CODE REVIEW

File: docs/articles/architecture/arch-vertical-configurations.rst [hunk 1]
```diff
@@ -10,11 +10,110 @@ Each vertical configuration describes a sub-system of the product.
 Update management
 =================
 
-The software update components are responsible for platform level software update.
+The software update components are responsible for platform level software
+update. This sub-system is critical in an automotive context since it allows to
+deploy new functionalities, fix important bugs and security issues or update
+assets such as driving data. A well designed update sub-system can save a lot
+of money by avoiding products recalls.
```
Comment (tobsan) [inline docs/articles/architecture/arch-vertical-configurations.rst]: I think we can add that not having to go to an auto repair shop to do this is a neat convenience plus probably a money saver.

File: docs/articles/architecture/arch-glossary.rst [hunk 2]
```diff
@@ -5,10 +5,22 @@
...
+| Kernel     | Refers to the Linux kernel in this document.   |
++------------+------------------------------------------------+
+| Rootfs     | Root File System (main partition mounted on /) |
++------------+------------------------------------------------+
+| ECU        | Electronic Control Unit                        |
++------------+------------------------------------------------+
+| TPM        | Trusted Platform Modules                       |
++------------+------------------------------------------------+
+| TEE        | ARM Trusted Execution Environment              |
++------------+------------------------------------------------+
```
Comment (tobsan) [inline docs/articles/architecture/arch-glossary.rst]: Please have a look at the `:glossary:` and `:term:` directives available in sphinx.

File: docs/articles/architecture/arch-vertical-configurations.rst [hunk 3]
```diff
@@ -10,11 +10,110 @@ Each vertical configuration describes a sub-system of the product.
...
 
 Use cases
 ---------
 
+A platform developer should be able to deploy rootfs, kernel, bootloader, ECUs,
```
Comment (tobsan) [inline docs/articles/architecture/arch-vertical-configurations.rst]: I do think that usually ECUs refer to the actual hardware, which the developer can't deploy. Perhaps the config of the ECU is what it should say?

File: docs/articles/architecture/arch-vertical-configurations.rst [hunk 4]
```diff
@@ -10,11 +10,110 @@ Each vertical configuration describes a sub-system of the product.
...
+ - Full downloads: alternatively, downloading complete images solves the issue
+   of local corrupted data but is much more resource intensive.
+
+On a different side, updates can also be file or block based which affect the
+portability of update data across devices and also the size of downloaded
+information.
+
+An OTA server should propose various fleet and deployment management scenarios.
+For example, a car manufacturer should be able to deploy updates geographically
+or following a schedule.
```
Comment (tobsan) [inline docs/articles/architecture/arch-vertical-configurations.rst]: And I suppose having the ability to only apply to some models in the fleet.

File: docs/articles/architecture/arch-vertical-configurations.rst [hunk 5]
```diff
@@ -10,11 +10,110 @@ Each vertical configuration describes a sub-system of the product.
...
+
+ - Integrity: this guarantees that data haven't been modified from the server
+   to the client. (for example, by man-in-the-middle attacks)
+
+ - Authentication: this guarantees that update have been created by an
+   authorized entity. (for example, by a Tier-1 or OEM vendor) This can be
+   achieved using digital signatures, CMAC or HMAC.
+
+In order to implement those checks correctly, it would be interesting to rely
+on a deep root of trust with TPM or TEE.
```
Comment (tobsan) [inline docs/articles/architecture/arch-vertical-configurations.rst]: Perhaps @e8johan can give some pointers on how this usually works?

**Review by tobsan — CHANGES_REQUESTED** [review]
Looks good in general

File: docs/articles/architecture/arch-glossary.rst (hunk 2 above)
Comment (GordanM) [reply docs/articles/architecture/arch-glossary.rst]: +1 it's my fault we've created the glossary in this way. Sorry!

**Review by GordanM — COMMENTED** [review]
Nice! I was wondering what would be a good spot to put the links to the online research previously done? Would it be this commit or rather https://github.com/Pelagicore/software-factory/pull/85/files ?

File: docs/articles/architecture/arch-vertical-configurations.rst (hunk 5 above)
Comment (FlorentRevest) [reply docs/articles/architecture/arch-vertical-configurations.rst]: I'll be waiting for his comments, if any.

**Comment by FlorentRevest** [comment]
PR Updated according to Tobias' comments.

@GordanM I don't get what "online research previously done" you're talking about.

**Comment by GordanM** [comment]
@FlorentRevest I was thinking about references to existing material - like presentations, case studies and so on.

**Comment by FlorentRevest** [comment]
Alright, will take care of that, thanks!

I would think that https://github.com/Pelagicore/software-factory/pull/85/files is more appropriate.

**Review by tobsan — APPROVED** [review]

**Pull request merged**